internal/scanner/scanner.go

```diff
@@ -71,6 +75,44 @@ func (s *Scanner) PreCheck(url string) (blocked bool, statusCode int, err error)
 	return blocked, code, nil
 }
 
+func (s *Scanner) WSPreCheck(url string) (available bool, blocked bool, err error) {
+	wsClient, _, err := s.wsClient.Dial(url, nil)
+	if err != nil {
+		return false, false, err
+	}
+
+	var WSpreCheckVectors = [...]string{
+		fmt.Sprintf("{\"message\": \"%[1]s\", \"%[1]s\": \"%[1]s\"}", preCheckVector),
+		preCheckVector,
+	}
+
+	wsError := make(chan error, 1)
+
+	go func() {
+		defer close(wsError)
+		for {
+			_, _, err := wsClient.ReadMessage()
+			if err != nil {
+				wsError <- err
+			}
+		}
+	}()
+
+	for _, payload := range WSpreCheckVectors {
+		select {
+			case err := <- wsError:
+				return true, true, err
```

Comment:
you are using the named variables available, blocked in function declaration but not in the body

Reply:
Fxd, they are not named now, because I don't use them as variables or don't use "naked" returns.